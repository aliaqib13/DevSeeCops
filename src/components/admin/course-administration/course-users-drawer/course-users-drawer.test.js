import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import CourseUsersDrawer from './index';

const props = {
    users: {
        data: [
            {
                activated: 1,
                certificate_name: null,
                coordinator: null,
                created_at: '2020-05-25 16:35:42',
                email: 'henry@araido.io',
                firstname: 'Henry',
                id: 3,
                lastname: 'Tovmasyan',
                updated_at: '2020-05-25 16:35:42',
            },
            {
                activated: 1,
                certificate_name: null,
                coordinator: null,
                created_at: '2020-05-25 16:35:42',
                email: 'george@araido.io',
                firstname: 'George',
                id: 4,
                lastname: 'Aramyan',
                updated_at: '2020-05-25 16:35:42',
            },
        ],
        lastPage: 1,
        page: 1,
        perPage: 10,
        total: 8,
    },
    visible: true,
    removeUserActiveCourse: jest.fn(() => Promise.resolve(true)),
    addUserActiveCourse: jest.fn(() => Promise.resolve(true)),
    onClose: jest.fn(),
    getCourseUsers: jest.fn(() => Promise.resolve(true)),
};

describe('CourseUsersDrawer', () => {
    let component;

    beforeEach(() => {
        component = shallow(<CourseUsersDrawer {...props} />);
    });

    it('Should render successfully', () => {
        const Drawer = component.find('withConfigConsumer(Drawer)');
        expect(Drawer.exists()).toBe(true);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('Should render successfully the autoComplete input', () => {
        const AutoComplete = component.find('AutoComplete');
        expect(AutoComplete.exists()).toBe(true);
    });

    it('Should render successfully the table containing users', () => {
        const table = component.find('withStore(Table)');
        expect(table.exists()).toBe(true);
        expect(table.props().dataSource).toEqual(props.users.data);
    });
});

describe('CourseUsersDrawer table', () => {
    let component;

    beforeEach(() => {
        component = mount(<CourseUsersDrawer {...props} />);
    });

    it('Should show user\'s email in users table', () => {
        const user_email = component.find('tr.drawer-users-row').at(0).find('td').at(0);
        expect(user_email.text()).toBe(props.users.data[0].email);
    });

    it('Should call removeUserActiveCourse function with proper user id when click on delete button in users table', () => {
        const remove_user_icon = component.find('tr.drawer-users-row').at(0).find('td').at(1)
            .find('Button');
        remove_user_icon.simulate('click');
        expect(props.removeUserActiveCourse).toBeCalledTimes(1);
        expect(props.removeUserActiveCourse.mock.calls[0][0]).toBe(props.users.data[0].id);
    });
});

describe('CourseUsersDrawer autoComplete Input', () => {
    let component; let
        Input;

    beforeEach(() => {
        component = shallow(<CourseUsersDrawer {...props} />);
        props.addUserActiveCourse.mockClear();
        Input = component.find('AutoComplete').find('Input');
    });

    it('Should call addUserActiveCourse when clicked on add_user icon and input field is not empty', () => {
        component.instance().setState({ inputValue: 'test@mail.test' });
        Input.props().suffix.props.onClick();
        expect(props.addUserActiveCourse).toBeCalledTimes(1);
    });

    it('Should not call addUserActiveCourse when clicked on add_user icon and input field is empty', () => {
        component.instance().setState({ inputValue: '' });
        Input.props().suffix.props.onClick();
        expect(props.addUserActiveCourse).toBeCalledTimes(0);
    });
});
