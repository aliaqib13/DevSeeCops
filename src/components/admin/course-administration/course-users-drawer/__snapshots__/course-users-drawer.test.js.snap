// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseUsersDrawer Should render successfully 1`] = `
<withConfigConsumer(Drawer)
  className="course-users-drawer"
  onClose={[MockFunction]}
  placement="right"
  switchScrollingEffect={false}
  title="Users"
  visible={true}
  width={550}
>
  <AutoComplete
    choiceTransitionName="zoom"
    dataSource={Array []}
    filterOption={false}
    onSearch={[Function]}
    onSelect={[Function]}
    optionLabelProp="children"
    placeholder="add by email"
    showSearch={false}
    size="large"
    style={
      Object {
        "width": "100%",
      }
    }
    transitionName="slide-up"
    value=""
  >
    <Input
      suffix={
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
          size="large"
          style={
            Object {
              "marginRight": -12,
            }
          }
          type="primary"
        >
          <Icon
            type="user-add"
          />
        </Button>
      }
      type="text"
    />
  </AutoComplete>
  <withStore(Table)
    dataSource={
      Array [
        Object {
          "activated": 1,
          "certificate_name": null,
          "coordinator": null,
          "created_at": "2020-05-25 16:35:42",
          "email": "henry@araido.io",
          "firstname": "Henry",
          "id": 3,
          "lastname": "Tovmasyan",
          "updated_at": "2020-05-25 16:35:42",
        },
        Object {
          "activated": 1,
          "certificate_name": null,
          "coordinator": null,
          "created_at": "2020-05-25 16:35:42",
          "email": "george@araido.io",
          "firstname": "George",
          "id": 4,
          "lastname": "Aramyan",
          "updated_at": "2020-05-25 16:35:42",
        },
      ]
    }
    pagination={
      Object {
        "current": 1,
        "defaultCurrent": 1,
        "onChange": [Function],
        "pageSize": 10,
        "position": "top",
        "total": 8,
      }
    }
    rowClassName="drawer-users-row"
    rowKey={[Function]}
    size="lg"
  >
    <Column
      dataIndex="email"
      key="email"
      title={[Function]}
    />
    <Column
      key="actions"
      render={[Function]}
      title="actions"
    />
  </withStore(Table)>
</withConfigConsumer(Drawer)>
`;
